docs/pages/guides/hello-world/add-chain-client.mdx

```diff
@@ -0,0 +1,93 @@
+import { CollapseCode } from "../../../components/CollapseCode";
+
+# Add chains to the client
+
+In this tutorial you modify a template application to use a public blockchain instead of [`anvil`](https://book.getfoundry.sh/anvil/).
+Note that this is not required if you are going to use Lattice chains, [redstone](https://redstone.xyz/) and [garnet](https://garnetchain.com/)
+
+## Setup
+
+[Create a new MUD application from the template](../../templates/typescript/getting-started).
+Use the vanilla template.
+
+```sh copy
+pnpm create mud@latest tutorial --template vanilla
+cd tutorial
+```
+
+You can also use the `react-ecs` or `threejs` templates without any changes to the tutorial.
+If you wish to use the `react` template you'll need to modify a bit to account for the use of Zustand instead of RECs.
+
+## File changes
+
+You need to edit both `supportedChains.ts` and `setupNetwork.ts`.
+
+- Edit the file `packages/client/src/mud/supportedChains.ts` to add the chain definition from Viem.
+  You can see [the list of chains supported by Viem](https://github.com/wevm/viem/blob/main/src/chains/index.ts).
+  If you need a different chain you can use [`defineChain`](https://viem.sh/docs/chains/introduction#custom-chains).
+
+  ```ts filename="supportedChains.ts" copy showLineNumbers {2,4}
+  import { MUDChain, mudFoundry, redstone, garnet } from "@latticexyz/common/chains";
+  import { holesky } from "viem/chains";
+
+  export const supportedChains: MUDChain[] = [mudFoundry, redstone, garnet, holesky];
+  ```
+
+- Very often chain objects in Viem don't have a webSocket endpoint, only an HTTP one.
+  Edit `packages/client/src/mud/setupNetwork.ts` to only use a webSocket endpoint when one is defined.
+
+  ```ts showLineNumbers {1-18,22}
+  const urlTypes =
+    (networkConfig.chain.rpcUrls.default.webSocket ? "w" : "-") +
+    (networkConfig.chain.rpcUrls.default.http ? "h" : "-");
+  let fallbacks = [];
+  switch (urlTypes) {
+    case "wh":
+      fallbacks = [webSocket(), http()];
+      break;
+    case "w-":
+      fallbacks = [webSocket()];
+      break;
+    case "-h":
+      fallbacks = [http()];
+      break;
+    case "--":
+    default:
+      console.log("No URL type I can use for the chain");
+  }
+
+  const clientOptions = {
+    chain: networkConfig.chain,
+    transport: transportObserver(fallback(fallbacks)),
+    pollingInterval: 1000,
+  } as const satisfies ClientConfig;
+  ```
```

Comment:
I would argue this is not something we have to explain how to make dynamic here. We can simplify to mentioning that depending on the chain you want to use you might have to adjust your `transport` client option for viem. Anything beyond that falls in the user's responsibility to customize / make dynamic as they wish.

Reply:
Removed, replaced with a much simpler note.